benchmark/solver/solver.cpp

```diff
@@ -170,6 +170,35 @@ const std::map<std::string, std::function<std::unique_ptr<gko::LinOpFactory>(
                    {"overhead", create_solver<gko::Overhead<>>}};
 
 
+// TODO: Workaround until GPU matrix conversions are implemented
+//       The factory will wrap another factory, and make sure that the
+//       input operator is copied to the reference executor, and then sent
+//       through the generate function
+struct ReferenceFactoryWrapper
+    : gko::EnablePolymorphicObject<ReferenceFactoryWrapper, gko::LinOpFactory> {
+    ReferenceFactoryWrapper(std::shared_ptr<const gko::Executor> exec)
+        : gko::EnablePolymorphicObject<ReferenceFactoryWrapper,
+                                       gko::LinOpFactory>(exec)
+    {}
+
+    ReferenceFactoryWrapper(std::shared_ptr<const gko::LinOpFactory> f)
+        : gko::EnablePolymorphicObject<ReferenceFactoryWrapper,
+                                       gko::LinOpFactory>(f->get_executor()),
+          base_factory{f}
+    {}
+
+    std::shared_ptr<const gko::Executor> exec{gko::ReferenceExecutor::create()};
+    std::shared_ptr<const gko::LinOpFactory> base_factory;
+
+protected:
+    std::unique_ptr<gko::LinOp> generate_impl(
+        std::shared_ptr<const gko::LinOp> op) const override
+    {
+        return base_factory->generate(gko::clone(exec, op));
+    }
+};
+
+
```

Comment:
Something may have gone wrong during the rebase, I removed this on purpose, since all matrix format conversions are now implemented on the GPU

Reply:
Yes, my bad, I did not pay close enough attention to the rebasing process.
Good catch!